### PR TITLE
Adds MIX Leave Element, its Parser and Serializer.

### PR DESCRIPTION
**Pull request by tarun018**
License:
This patch is BSD-licensed, see Documentation/Licenses/BSD-simplified.txt for details.

Test-Information:
Added tests for MIX Leave Parser and Serializer based on examples in XEP 0369, which passes.

Change-Id: Ie9eb3341b23099de4dda2b9915d28115dbfe5837

### CODE REVIEW

File: Swiften/Elements/MIXLeave.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+#pragma once
+
+#include <memory>
+
+#include <boost/optional.hpp>
+
+#include <Swiften/Base/API.h>
+#include <Swiften/Base/Override.h>
```
Comment (tfar) [inline Swiften/Elements/MIXLeave.h]: This include isn't used.

**Comment by tarun018** [comment]
Updated based on feedback.

**Comment by tfar** [comment]
The superfluous #include statement is still in there, not?

**Comment by tarun018** [comment]
Removed Override from Element and added to Parser header file.

File: Swiften/Parser/PayloadParsers/MIXLeaveParser.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * Copyright (c) 2017 Tarun Gupta
+ * Licensed under the simplified BSD license.
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#include <Swiften/Parser/PayloadParsers/MIXLeaveParser.h>
+
+#include <boost/optional.hpp>
+
+using namespace Swift;
```
Comment (tfar) [inline Swiften/Parser/PayloadParsers/MIXLeaveParser.cpp]: Please use `namespace Swift {` in implementation files instead.

File: Swiften/Serializer/PayloadSerializers/MIXLeaveSerializer.cpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#include <Swiften/Serializer/PayloadSerializers/MIXLeaveSerializer.h>
+
+#include <memory>
+
+#include <Swiften/Serializer/XML/XMLElement.h>
+
+using namespace Swift;
```
Comment (tfar) [inline Swiften/Serializer/PayloadSerializers/MIXLeaveSerializer.cpp]: Please use `namespace Swift {` instead.

**Comment by tfar** [comment]
Also please rebase so that it merges without conflict.

**Comment by tarun018** [comment]
Updated and rebased.

**Comment by tfar** [comment]
Thank you for this contribution. This change has been submitted and merged upstream.

**Pull request closed without merging**